src/command.rs

```diff
@@ -309,3 +333,59 @@ impl File {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct Command {
+    index: usize,
+    main: String,
+    detail: Vec<String>,
+}
+
+impl Command {
+    pub fn new(detail: Vec<String>) -> Command {
+        let mut d_iter = detail.iter();
+        let mut detail = Vec::new();
+        let main = match d_iter.next() {
+            Some(cmd) => cmd.to_string(),
+            None => "/bin/bash".to_string(),
+        };
+
+        for d in d_iter {
+            detail.push(d.to_string())
+        }
+
+        Command {
+            index: 0,
+            main: main,
+            detail: detail,
+        }
+    }
+
+    pub fn main(&self) -> &str {
+        &self.main
+    }
+
+    // pub fn detail_iter(&self) -> Vec<&str> {
+    //     let res: Vec<&str> = self.detail.iter().map(|d| &(*d)).collect::<Vec<&str>>();
+    //     res
+    // }
+
+    pub fn detail(&self) -> &Vec<String> {
+        &self.detail
+    }
+}
```

Comment:
構造体の名前が`Command`だと、このモジュールの根幹についてのものかと思ってしまう(が、commandモジュールはinjeshのサブコマンドを扱うモジュールである)。また、`std::command::Command`とダブるので勘違いしやすい。

変数名を読みやすく変更

イテレータの全要素に単純な操作をするなら、for文よりmapのが適切と思う

```suggestion
#[derive(Debug)]
pub struct ExecuteCommand {
    index: usize,
    main: String,
    args: Vec<String>,
}

impl ExecuteCommand {
    pub fn new(raw_commands: Vec<String>) -> ExecuteCommand {
        let mut raw_commands_iter = raw_commands.iter();
        let main = match raw_commands_iter.next() {
            Some(cmd) => cmd.to_string(),
            None => "/bin/bash".to_string(),
        };

        let args: Vec<String> = raw_commands_iter.map(|arg| arg.to_string()).collect();

        ExecuteCommand {
            index: 0,
            main: main,
            args: args,
        }
    }

    pub fn main(&self) -> &str {
        &self.main
    }

    pub fn args(&self) -> &Vec<String> {
        &self.args
    }
}
```

Reply:
`index`は何に使う予定なの？